apps/braze/functions/updateContentBlocks.ts

```diff
@@ -4,33 +4,143 @@ import {
   FunctionEventContext,
   AppEventRequest,
 } from '@contentful/node-apps-toolkit';
+import {
+  getConfigEntry,
+  CONFIG_FIELD_ID,
+  CONFIG_ENTRY_ID,
+  CONFIG_CONTENT_TYPE_ID,
+  updateConfig,
+} from '../src/utils';
+import { createClient, EntryProps, KeyValueMap, PlainClientAPI } from 'contentful-management';
+import { documentToHtmlString } from '@contentful/rich-text-html-renderer';
+
+const WAIT_TIMES = [0, 5000, 10000];
+
+type EntryConnectedFields = {
+  fieldId: string;
+  contentBlockId: string;
+}[];
+
+type AppInstallationParameters = {
+  brazeApiKey: string;
+  brazeEndpoint: string;
+};
+
+function initContentfulManagementClient(context: FunctionEventContext): PlainClientAPI {
+  if (!context.cmaClientOptions) {
+    throw new Error(
+      'Contentful Management API client options are only provided for certain function types. To learn more about using the CMA within functions, see https://www.contentful.com/developers/docs/extensibility/app-framework/functions/#using-the-cma.'
+    );
+  }
+  return createClient(context.cmaClientOptions, {
+    type: 'plain',
+    defaults: {
+      spaceId: context.spaceId,
+      environmentId: context.environmentId,
+    },
+  });
+}
 
-/**
- * App Event Handler Function Template
- *
- * This template provides the structure for creating a handler function
- * for Contentful app events.
- *
- * Customize the implementation within the handler
- * to define your specific handling logic.
- *
- * @param event - The app event to be handled
- * @param context - The execution context
- * @returns None - App event handlers don't return a response
- */
 export const handler: FunctionEventHandler<FunctionTypeEnum.AppEventHandler> = async (
   event: AppEventRequest,
   context: FunctionEventContext
 ) => {
-  // Check if the event is an Entry event
-  // You can also handle other event types like Asset or ContentType
-  if (event.headers['X-Contentful-Topic'].includes('Entry')) {
-    const entryEvent = event.body;
+  const { brazeApiKey, brazeEndpoint } =
+    context.appInstallationParameters as AppInstallationParameters;
+  const cma = initContentfulManagementClient(context);
+
+  if (event.headers['X-Contentful-Topic'].includes('AppInstallation.delete')) {
+    await callAndRetry(() => deleteConfigEntry(cma));
+    return;
+  }
+
+  const body = event.body as any;
+  const entryId = body.sys.id;
+
+  const configEntry: EntryProps<KeyValueMap> = await getConfigEntry(cma);
+  const configField = configEntry.fields[CONFIG_FIELD_ID];
+  if (!configField) {
+    return;
+  }
```

Comment:
💭 I'd add a console.error("Not able to find {CONFIG_FIELD_ID}") here so if it fails, we can see it in the function's logs